zulipterminal/model.py

```diff
@@ -393,6 +393,19 @@ def toggle_message_reaction(
             response = self.client.add_reaction(reaction_to_toggle_spec)
         display_error_if_present(response, self.controller)
 
+    def has_user_reacted_to_msg(self, emoji_name: str, message: Message) -> bool:
+        for reaction in message["reactions"]:
+            if reaction["emoji_name"] != emoji_name:
+                continue
+            user = reaction["user"]
+            has_user_reacted = (
+                user.get("user_id", None) == self.user_id
+                or user.get("id", None) == self.user_id
```

Comment:
Right, I see they introduced the `dict(user_id=...)` in server version 3.0 ZFL 2. So that means, we'd still need to support the above along with the new unified format?

Reply:
@neiljp As discussed in the stream, I have kept all three check for getting "user_id", this should hopefully survive post the removal of `user` object from the server.

I have also centralized the logic for `has_user_reacted_to_message` and adapted the toggle-reaction logic to use this method instead. As you said, this should also prove useful in the display logic. :+1: